subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/BuildScanPluginSmokeTest.groovy

```diff
@@ -156,6 +201,36 @@ class BuildScanPluginSmokeTest extends AbstractSmokeTest {
         version << UNSUPPORTED
     }
 
+    @Requires(IntegTestPreconditions.NotConfigCached)
+    def "can inject plugin #pluginVersion in #ci using '#ciScriptVersion' script version"() {
+        def versionNumber = VersionNumber.parse(pluginVersion)
+        def initScript = "init-script.gradle"
+        file(initScript) << getCiInjectionScriptContent(ci)
+
+        // URL is not relevant as long as it's valid due to the `-Dscan.dump` parameter
+        file("gradle.properties") << """
+            systemProp.${ci.propPrefix}.gradle-enterprise.plugin.version=$pluginVersion
+            systemProp.${ci.propPrefix}.init-script.name=$initScript
+            systemProp.${ci.propPrefix}.gradle-enterprise.url=http://localhost:5086
+        """.stripIndent()
+
+        setupJavaProject()
+
+        expect:
+        scanRunner("--init-script", initScript)
+            .expectLegacyDeprecationWarningIf(versionNumber < FIRST_VERSION_CALLING_BUILD_PATH,
+                "The BuildIdentifier.getName() method has been deprecated. " +
+                    "This is scheduled to be removed in Gradle 9.0. " +
+                    "Use getBuildPath() to get a unique identifier for the build. " +
+                    "Consult the upgrading guide for further information: https://docs.gradle.org/${GradleVersion.current().version}/userguide/upgrading_version_8.html#build_identifier_name_and_current_deprecation"
+            )
+            .build().output.contains("Build scan written to")
+
+        where:
+        [ci, pluginVersion] << [CI.values(), SUPPORTED_BY_CI_INJECTION].combinations()
```

Comment:
❓ How long does this test take if we run it for all the supported plugin versions? Is it long enough it would make sense to reduce the number of combinations?